pallets/multisig/src/lib.rs

```diff
@@ -209,16 +228,24 @@ decl_module! {
             use sp_version::RuntimeVersion;
             use polymesh_primitives::migrate::kill_item;
 
-            // Kill pending proposals if the transaction version is upgraded
-            let current_version = <T::Version as Get<RuntimeVersion>>::get().transaction_version;
-            let last_version = TransactionVersion::get();
-            if last_version < current_version {
-                TransactionVersion::set(current_version);
-                for item in &["Proposals", "ProposalIds", "ProposalDetail", "Votes"] {
-                    kill_item(b"MultiSig", item.as_bytes())
+            let storage_ver = <StorageVersion>::get();
+
+            // Migration from V0 to V1.
+            if storage_ver == Version::V0 {
+                // Kill pending proposals if the transaction version is upgraded
+                let current_version = <T::Version as Get<RuntimeVersion>>::get().transaction_version;
+                let last_version = TransactionVersion::get();
+                if last_version < current_version {
+                    TransactionVersion::set(current_version);
+                    for item in &["Proposals", "ProposalIds", "ProposalDetail", "Votes"] {
+                        kill_item(b"MultiSig", item.as_bytes())
+                    }
                 }
+
+                <StorageVersion>::put(Version::V1);
             }
```

Comment:
This "migration" needs to run on every upgrade. It has its own logic of checking if the migration is required `(last_version < current_version)`